### ISSUE TITLE
[Bug]: freetube wont load

### ISSUE THREAD

**Issue by ticmaidev** [FreeTubeApp/FreeTube#4223]
### Guidelines

- [X] I have encountered this bug in the [latest release of FreeTube](https://github.com/FreeTubeApp/FreeTube/releases).
- [X] I have encountered this bug in the [official downloads of FreeTube](https://github.com/FreeTubeApp/FreeTube#official-downloads).
- [X] I have searched the issue tracker for [open](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aopen+is%3Aissue) and [closed](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aissue+is%3Aclosed) issues that are similar to the bug report I want to file, without success.
- [X] I have searched the [documentation](https://docs.freetubeapp.io/) for information that matches the description of the bug I want to file, without success.
- [X] This issue contains only one bug.

### Describe the bug

1. just open freetube
2. blank screen forever

### Expected Behavior

it should show my feed?

### Issue Labels

content not loading

### FreeTube Version

v0.19.1 Beta

### Operating System Version

Ubuntu 20.04.6 LTS

### Installation Method

.deb

### Primary API used

Local API

### Additional Information

I've tried downgrading to v0.18.x but it won't work, as well...
![image](https://github.com/FreeTubeApp/FreeTube/assets/128381407/1bd16d40-1dd6-4dcc-8924-a0b08cd075c8)
![image](https://github.com/FreeTubeApp/FreeTube/assets/128381407/6cae84d9-80e2-4501-b74b-847b9fc3546b)

### Nightly Build

- [ ] I have encountered this bug in the latest [nightly build](https://docs.freetubeapp.io/development/nightly-builds).

**Comment by ticmaidev** [comment]
invidious api instances endpoint doesn't seem to be working, since the request never returns anything (times out).

could this be the cause of this issue?

![image](https://github.com/FreeTubeApp/FreeTube/assets/128381407/1e761b08-1e02-4376-8417-5546d5fa4ce9)

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Hi could u try the latest [nightly build](https://docs.freetubeapp.io/development/nightly-builds) and report back

**Comment by ticmaidev** [comment]
> Hi could u try the latest [nightly build](https://docs.freetubeapp.io/development/nightly-builds) and report back

same problem.

**Comment by ticmaidev** [comment]
@efb4f5ff-1298-471a-8973-3d47447115dc it's definitely invidious fault. the api endpoint got back to working and freetube finally loaded.

maybe freetube shoul've had a fallback for this case.

**Comment by OTKUSteyler** [comment]
i have a simuler probelm but freetube won't show no ui at all

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
> Hi could u try the latest [nightly build](https://docs.freetubeapp.io/development/nightly-builds) and report back

@OTKUSteyler have u tries this?

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@ticmaidev to my knowledge Invidious shouldn't impact FT like this. I'll check with the team to make sure i didnt miss something

**Comment by OTKUSteyler** [comment]
> > Hi could u try the latest [nightly build](https://docs.freetubeapp.io/development/nightly-builds) and report back
> 
> @OTKUSteyler have u tried this?
 
 yes i did try it and every time is "runs,, in the background and there is no ui after all

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Idk If this is even possible but could u both open the console to check if there are any errors in there.

Ctrl + shift + i

**Comment by OTKUSteyler** [comment]
> Idk If this is even possible but could u both open the console to check if there are any errors in there.
> 
> Ctrl + shift + i

![Screenshot_20231023_202914](https://github.com/FreeTubeApp/FreeTube/assets/90369564/b5e6c536-4e0a-43ec-bbe0-8edf1e57d7d8)

i sadly can't see nothing be is  running  and not even Ctrl + Shift + i is just not  working cus i can't see nothing

 this is the only thing wich i can only see

**Comment by ticmaidev** [comment]
@efb4f5ff-1298-471a-8973-3d47447115dc 

it was said invidious instances API endpoint for my scenario, indeed.
I was able to reproduce the issue by changing some lines to simulate invidious taking too long to respond:

```js
// src/renderer/store/modules/invidious.js

async function fetchDataAfterDelay(requestUrl) {
  await new Promise((resolve) => {
    setTimeout(resolve, 1000 * 60 * 2); // 2 minutes delay (2 minutes * 60 seconds * 1000 milliseconds)
  });

  return await fetch(requestUrl)
}

const actions = {
  async fetchInvidiousInstances({ commit }) {
    const requestUrl = 'https://api.invidious.io/instances.json'

    let instances = []
    try {
      const response = await fetchDataAfterDelay(requestUrl)
    // continues...
}
```


```js
// _scripts/getInstances.js

const fs = require('fs/promises')
const invidiousApiUrl = 'https://api.invidious.io/instances.json'

setTimeout(() => {
  fetch(invidiousApiUrl)
    .then((e) => e.json())
    .then((res) => {
      const data = res.filter((instance) => {
        return !(
          instance[0].includes('.onion') ||
          instance[0].includes('.i2p') ||
          !instance[1].api
        );
      }).map((instance) => {
        return {
          url: instance[1].uri.replace(/\/$/, ''),
          cors: instance[1].cors,
        };
      });
      fs.writeFile('././static/invidious-instances.json', JSON.stringify(data, null, 2));
    });
}, 1000 * 60 * 2); // 2 minutes delay (2 minutes * 60 seconds * 1000 milliseconds)
```

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Retrieved from dev chat

> looks like when the invidious instances api is down freetube doesn't time out quick enough so it just sits on a blank screen

> which means there are two issues, firstly we need to add a timeout that times out faster, also elements that don't need invidious like the top bar should not wait until invidious responds to load in

**Comment by ticmaidev** [comment]
@efb4f5ff-1298-471a-8973-3d47447115dc 

are you an ai bot or something? curious nickname

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Ha, no ai, just somebody that didnt care about their username :)
